src/Lib.P2P/Routing/DhtService.cs

```diff
@@ -0,0 +1,444 @@
+
+                _log.Debug($"got {request.Type} from {connection.RemotePeer}");
+                var response = new DhtMessage
+                {
+                    Type = request.Type,
+                    ClusterLevelRaw = request.ClusterLevelRaw
+                };
+                switch (request.Type) 
+                {
+                    case MessageType.Ping:
+                        response = ProcessPing(request, response);
+                        break;
+                    case MessageType.FindNode:
+                        response = ProcessFindNode(request, response);
+                        break;
+                    case MessageType.GetProviders:
+                        response = ProcessGetProviders(request, response);
+                        break;
+                    case MessageType.AddProvider:
+                        response = ProcessAddProvider(connection.RemotePeer, request, response);
+                        break;
+                    case MessageType.PutValue:
+                        break;
+                    case MessageType.GetValue:
+                        break;
+                    default:
+                        _log.Debug($"unknown {request.Type} from {connection.RemotePeer}");
+
+                        // TODO: Should we close the stream?
+                        continue;
+                }
+
+                if (response == null)
+                {
+                    continue;
+                }
+                
+                Serializer.SerializeWithLengthPrefix(stream, response, PrefixStyle.Base128);
+                await stream.FlushAsync(cancel).ConfigureAwait(false);
+            }
+        }
+
+        /// <inheritdoc />
+        public Task StartAsync()
+        {
+            _log.Debug("Starting");
+
+            RoutingTable = new RoutingTable(SwarmService.LocalPeer);
+            ContentRouter = new ContentRouter();
+            SwarmService.AddProtocol(this);
+            SwarmService.PeerDiscovered += Swarm_PeerDiscovered;
+            SwarmService.PeerRemoved += Swarm_PeerRemoved;
+            
+            foreach (var peer in SwarmService.KnownPeers)
+            {
+                RoutingTable.Add(peer);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        /// <inheritdoc />
+        public Task StopAsync()
+        {
+            _log.Debug("Stopping");
+
+            SwarmService.RemoveProtocol(this);
+            SwarmService.PeerDiscovered -= Swarm_PeerDiscovered;
+            SwarmService.PeerRemoved -= Swarm_PeerRemoved;
+
+            Stopped?.Invoke(this, EventArgs.Empty);
+            ContentRouter?.Dispose();
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        ///   The swarm has discovered a new peer, update the routing table.
+        /// </summary>
+        private void Swarm_PeerDiscovered(object sender, Peer e)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unused method parameter 'sender'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)